app/workers/proactive_project_sync_worker.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class ProactiveProjectSyncWorker
+  include Sidekiq::Worker
+  sidekiq_options queue: :small, lock: :until_executed
+
+  PLATFORMS = %w[NPM Maven].freeze
+  DEFAULT_LIMIT = 1000
+
+  def perform(limit = DEFAULT_LIMIT)
+    projects_query
+      .limit(limit)
+      .each(&:async_sync)
+  end
+
+  def projects_query
+    Project
+      .distinct
+      .visible
+      .platform(PLATFORMS)
+      .where.associated(:repository_maintenance_stats)
+      .where("projects.last_synced_at IS NULL OR projects.last_synced_at < ?", last_synced_at_cutoff)
+      .order("projects.last_synced_at ASC NULLS FIRST")
```

Comment:
Are we guaranteed that this always gets populated? Would like to avoid repeated polling of null last_synced_ats.

Reply:
No, it can be null. There's only ~1,000 from this set that are null, but I don't think it's a bad state

Reply:
As long as we are not always filling up the first ~1000 slots with the same packages every time

Reply:
I don't believe this will be the case now, but I do agree it would be nice to have some warning if/when this starts happening. We should see fewer nulls and the mean last_synced_at for this set increase (while accounting for the passage of time)